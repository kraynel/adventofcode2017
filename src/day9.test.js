const { removeEscapes, removeGarbage, scoreGroups } = require("./day9");
const fs = require("fs");

describe("removeEscapes", () => {
  it("should remove escapes", () => {
    expect(removeEscapes("<{!>}>")).toBe("<{}>");
    expect(removeEscapes("<!!>")).toBe("<>");
    expect(removeEscapes("<!!!>>")).toBe("<>");
    expect(removeEscapes('<{o"i!a,<{i<a>')).toBe('<{o"i,<{i<a>');
  });
});

describe("removeGarbage", () => {
  it("should remove garbage", () => {
    expect(removeGarbage("<{!>}>")).toBe("");
    expect(removeGarbage("<!!>")).toBe("");
    expect(removeGarbage("<!!!>>")).toBe("");
    expect(removeGarbage('<{o"i!a,<{i<a>')).toBe("");

    expect(removeGarbage("{}")).toBe("{}");
    expect(removeGarbage("{{{}}}")).toBe("{{{}}}");
    expect(removeGarbage("{{},{}}")).toBe("{{},{}}");
    expect(removeGarbage("{{{},{},{{}}}}")).toBe("{{{},{},{{}}}}");
    expect(removeGarbage("{<{},{},{{}}>}")).toBe("{}");
    expect(removeGarbage("{<a>,<a>,<a>,<a>}")).toBe("{,,,}");
    expect(removeGarbage("{{<a>},{<a>},{<a>},{<a>}}")).toBe("{{},{},{},{}}");
    expect(removeGarbage("{{<!>},{<!>},{<!>},{<a>}}")).toBe("{{}}");
  });
});

describe("scoreGroups", () => {
  it("should score the groups", () => {
    expect(scoreGroups("{}")).toBe(1);
    expect(scoreGroups("{{{}}}")).toBe(6);
    expect(scoreGroups("{{},{}}")).toBe(5);
    expect(scoreGroups("{{{},{},{{}}}}")).toBe(16);
    expect(scoreGroups("{<{},{},{{}}>}")).toBe(1);
    expect(scoreGroups("{<a>,<a>,<a>,<a>}")).toBe(1);
    expect(scoreGroups("{{<ab>},{<ab>},{<ab>},{<ab>}}")).toBe(9);
    expect(scoreGroups("{{<!!>},{<!!>},{<!!>},{<!!>}}")).toBe(9);
    expect(scoreGroups("{{<a!>},{<a!>},{<a!>},{<ab>}}")).toBe(3);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day9.txt", "utf8", (err, data) => {
      expect(scoreGroups(data)).toMatchSnapshot();
      done();
    });
  });
});
