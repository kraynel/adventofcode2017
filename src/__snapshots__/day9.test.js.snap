// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scoreGroups should get the value 1`] = `14204`;
